moveit_planners/pilz_industrial_motion_planner/src/command_list_manager.cpp

```diff
@@ -112,7 +112,24 @@ RobotTrajCont CommandListManager::solve(const planning_scene::PlanningSceneConst
                               // therefore: "i-1".
                               (i > 0 ? radii.at(i - 1) : 0.));
   }
-  return plan_comp_builder_.build();
+
+  const auto res_vec = plan_comp_builder_.build();
+
+  // De-duplicate trajectory points with the same time value.
+  // This is necessary since some controllers do not allow times that are not monotonically increasing.
```

Comment:
// De-duplicate trajectory points with the same time value.
  // This is necessary since some controllers do not allow times that are not monotonically increasing.
  // TODO: Ideally, we would not need this code if the trajectory segments were created without
  // duplicate time points in the first place. Leaving this note to revisit this with a more principled fix.